### PR TITLE
icecap: Incorporate upstream build system adaptations into Veracruz

### PR DESCRIPTION
**Pull request by nspin**
IceCap has seen a number of recent changes and improvements, mostly with Veracruz in mind. This PR incorporates them into Veracruz.

The main change is that all of the IceCap source code is checked into the IceCap repository. That means that we can eliminate the complexity from the Veracruz build system for IceCap that was caused by the fact that IceCap Cargo manifests were generated at build time.

I've switched to including IceCap as a submodule.

### CODE REVIEW

**Review by dominic-mulligan-arm — COMMENTED** [review]
LGTM, though to double check: @nspin have you run `rustfmt` on all of the Rust source files and run the TOML tidying procedure on the `Cargo.toml` files?

**Comment by nspin** [comment]
@dominic-mulligan-arm The changes to Rust and Cargo files in this PR are trivial, save for the addition of a Cargo.toml file [1]. I haven't been able to find the Cargo.toml format checking script in the repo (I thought it would be in ./githooks). How can I run the check?

https://github.com/veracruz-project/veracruz/pull/259/files?authenticity_token=KkA7CyJRpBoUI80h%2FeK6nQBONKakPtLACUVkGkzti6AWhfrW%2FcnoEFcB5Y%2FKdWDHFZK2PZbVhSI8FTfdf5Fi0g%3D%3D&file-filters%5B%5D=.rs&file-filters%5B%5D=.toml&hide-deleted-files=true

**Comment by dominic-mulligan-arm** [comment]
Actually, @nspin just ignore the TOML formatting aspect of my previous message.

**Comment by nspin** [comment]
In that case, this PR is all set on that front. `rustfmt` accepts the `.rs` changes here.

**Review by dominic-mulligan-arm — APPROVED** [review]

File: icecap/Makefile [hunk 1]
```diff
@@ -99,7 +78,7 @@ test-system-stale: $(now) # implicit prerequisites: runtime-manager veracruz-ser
 set-key-permissions:
 	chmod 0400 nix/host/token-ssh-key.priv
 
-run_tests_body = eval $$($(nix_build) $(nix_expression) -A $(nix_attribute_path).run-tests --no-out-link)
+run_tests_body = script=$$($(nix_build) $(nix_expression) -A $(nix_attribute_path).run-tests --no-out-link) && eval $$script
```
[the remark this answers is not in the document]
Comment (nspin) [reply icecap/Makefile:81]: Due to some details of the default `sh` error handling configuration, in the red case, if overall shell fragment succeeds even if the subshell fails. In the green case, if fragment fails if the subshell fails, which is the desired behavior.

**Pull request merged**